File: KantaroBLL/Modi/ModoMinore.vb
Public Class ModoMinoreNaturale
    Inherits ModoBase

    Private Sub New()
        CreaSuccessioneDistanze(New Generic.List(Of Distanza)({Tono.Value,
                                                                Semitono.Value,
                                                                Tono.Value,
                                                                Tono.Value,
                                                                Semitono.Value,
                                                                Tono.Value,
                                                                Tono.Value}))
    End Sub

    Public Shared Function GetInstance() As ModoMinoreNaturale
        Return New ModoMinoreNaturale
    End Function
End Class

Public Class ModoMinoreMelodica
    Inherits ModoBase

    Private Sub New()
        CreaSuccessioneDistanze(New Generic.List(Of Distanza)({Tono.Value,
                                                              Semitono.Value,
                                                              Tono.Value,
                                                              Tono.Value,
                                                              Tono.Value,
                                                              Tono.Value,
                                                              Semitono.Value}))
    End Sub

    Public Shared Function GetInstance() As ModoMinoreMelodica
        Return New ModoMinoreMelodica
    End Function
End Class

Public Class ModoMinoreArmonica
    Inherits ModoBase

    Private Sub New()
        CreaSuccessioneDistanze(New Generic.List(Of Distanza)({Tono.Value,
                                                              Semitono.Value,
                                                              Tono.Value,
                                                              Tono.Value,
                                                              Semitono.Value,
                                                              Tono.Value + Semitono.Value,
                                                              Semitono.Value}))
    End Sub

    Public Shared Function GetInstance() As ModoMinoreArmonica
        Return New ModoMinoreArmonica
    End Function
End Class

File: Enzo.Music/Modi/ModoMinore.cs
namespace Enzo.Music;

public class ModoMinoreNaturale : ModoBase
{
    public ModoMinoreNaturale()
    {
        CreaSuccessioneDistanze(new List<Distanza>()
            {
                Tono.Value,
                Semitono.Value,
                Tono.Value,
                Tono.Value,
                Semitono.Value,
                Tono.Value,
                Tono.Value
            });
    }
    public static ModoMinoreNaturale GetInstance() => new();
}

public class ModoMinoreMelodica : ModoBase
{
    public ModoMinoreMelodica()
    {
        CreaSuccessioneDistanze(new List<Distanza>()
            {
                Tono.Value,
                Semitono.Value,
                Tono.Value,
                Tono.Value,
                Tono.Value,
                Tono.Value,
                Semitono.Value
            });
    }
    public static ModoMinoreMelodica GetInstance() => new();
}

public class ModoMinoreArmonica : ModoBase
{
    public ModoMinoreArmonica()
    {
        CreaSuccessioneDistanze(new List<Distanza>()
            {
                Tono.Value,
                Semitono.Value,
                Tono.Value,
                Tono.Value,
                Semitono.Value,
                Tono.Value + Semitono.Value,
                Semitono.Value
            });
    }
    public static ModoMinoreArmonica GetInstance() => new();
}
